### PR TITLE
Improvements for new prototype

### PR DESCRIPTION
**Pull request by chasehensel**
@awans most substantial is the improvement to the client api. Check it out!

### CODE REVIEW

**Comment by chasehensel** [comment]
@awans take a look whenever! This now contains explorer which you can run locally (though we'd have to change where the api points to to use it)

File: .gitignore [hunk 1]
```diff
@@ -22,4 +22,7 @@ aft
 
 #client
 /client/catalog/node_modules/
+/client/catalog/package-lock.json
```
Comment (awans) [inline .gitignore:25]: are we supposed to gitignore this?

File: README.md [hunk 2]
```diff
@@ -7,8 +7,14 @@ Run the server:
 
 `go run ./cmd/aft -db <path_to_db_file>`
 
-Run the client:
```
Comment (awans) [inline README.md:10]: nice!! thanks for updating docs :)

File: client/catalog/src/app/datatypes/DatatypeDetail.svelte [hunk 3]
```diff
@@ -9,12 +9,13 @@ import HLRowButton from '../../ui/HLRowButton.svelte';
...
 import HLCodeMirror from '../../ui/HLCodeMirror.svelte';
 import HLRow from '../../ui/HLRow.svelte';
+import HLDetailRow from '../../ui/HLDetailRow.svelte';
 import HLTextBig from '../../ui/HLTextBig.svelte';
 import HLSelect from '../../ui/HLSelect.svelte';
 import {router} from '../router.js';
 
 let id = params.id;
-let load = client.datatype.findMany({
+let load = client.api.datatype.findMany({
```
Comment (awans) [inline client/catalog/src/app/datatypes/DatatypeDetail.svelte:18]: +1 to this change!

File: client/catalog/src/app/datatypes/DatatypeDetail.svelte [hunk 4]
```diff
@@ -58,11 +59,19 @@ function setUpCM() {
 }
 
 async function updateDatatype() {
+	if(dt.enum == false && cm != null) {
+		const parses = await client.rpc.parse({data: {data : cm.getValue()}});
```
Comment (awans) [inline client/catalog/src/app/datatypes/DatatypeDetail.svelte:63]: cool!!

Comment (awans) [reply client/catalog/src/app/datatypes/DatatypeDetail.svelte:63]: I love RPCs!

File: client/catalog/src/data/client.js [hunk 5]
```diff
@@ -1,80 +1,48 @@
...
+  const responseBody = await res.json();
+  if ("code" in responseBody) {
+    return Promise.reject(responseBody);
+  }
+  return responseBody.data;
 }
 
-var client = new HttpRpcClient("https://localhost:8080/", CoreApi);
+var client = {
+  api: new Proxy(
```
Comment (awans) [inline client/catalog/src/data/client.js:19]: 👨‍🍳 👌

Comment (awans) [reply client/catalog/src/data/client.js:19]: (this is the best "chef's kiss" i can do until github ships that emoji)

File: client/explorer/public/index.html [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="shortcut icon" href="https://cdn.glitch.com/c5d47192-44d0-4954-8e72-ad5c9fd4601c%2Ffavicon.ico?v=1593736569935" />
```
Comment (awans) [inline client/explorer/public/index.html:5]: maybe embed the favicon in our codebase

File: internal/rpc/models.go [hunk 7]
```diff
@@ -34,27 +35,132 @@ var reactFormRPC = db.Code{
...
+result(validate(args))`,
 }
+
+var replRPC = db.Code{
+	ID:                db.MakeID("591bc8f7-543b-4fa9-bdf7-8948c79cdd26"),
+	Name:              "repl",
+	Runtime:           db.Starlark,
+	FunctionSignature: db.RPC,
+	Code: `def repl(args):
+    out, ran = Exec(args["data"], "")
```
Comment (awans) [inline internal/rpc/models.go:146]: I realize we're early here, but I think we should really consider the security implications (alarm bells!!) of having something like this in the code.

Comment (awans) [reply internal/rpc/models.go:146]: I realize the REPL is basically just "SQL Injection as a Feature", so what I have in mind here is at least some kind of authentication -- which I realize you're blocked on me for :p 

so maybe just add some big scary comments around this for now!

File: .gitignore (hunk 1 above)
Comment (chasehensel) [reply .gitignore:25]: They are huge, and get created on npm install. If I don't git ignore, then explorer has a 15k long file that will be added. Seemed silly

**Pull request merged**